### ISSUE TITLE
Enforce PRs to have at least one of the labels `bug`, `enhancement`, or `interface-change`

### ISSUE THREAD

**Issue by jesusvasquez333** [slaclab/pysmurf#321]
Setup a GitHub workflow to enforce the use of one of these mandatory labels:
- `bug`
- `enhancement`
- `interface-change`

before merging a PR. 

These labels will be used to autogenerate release notes

**Comment by jesusvasquez333** [comment]
I did some research and some testing regarding this feature and unfortunately I don't think there is a convenient way to implement this. 

One option I tested was to use travis to run a python script too check if the PR has one of the mandatory labels. In travis a stage to run the script will be trigger by the condition `if: type = pull_request` to make sure the test runs only PRs. However, the issue is that travis is only triggered when the PR is created, and with every new push to the PR. This means, for example, that is one opens a PR without adding a label initially, then the travis test will fail, but adding a label later won't trigger travis again to update the result of the test; only doing a new push will do it, which is not what we want.

Another option is to use a Github workflow. Theses have the advantage that they can be trigger both when a PR is crated as well as when a label is added to the PR. Also, I found a GitHub action in the marketplace called [enforce-pr-label](https://github.com/marketplace/actions/enforce-pr-label) which does exactly what we need, verify that a PR has a proper label. However, during my tests I found out that the two trigger conditions (opening a PR, and adding a label to a PR) will create 2 independent checks in GitHub. This means that, if for example one opens a PR without a label, then a GitHub check will fail; then adding a label will start a second Github check that will succeed, but the first check will remain in the failed state, and the PR's merge will be block. 

Additionally to theses issues, there is a more fundamental issue: even if we enforce a PR to have labels before they re merge, labels on a merged PR can be change anyways. So, it is unlikely but it is possible that an user remove a label of a merged PR. Then, at the time to generate the release notes, the PR could be missing the label we need.

**Comment by slacrherbst** [comment]
Ok. We will just be sure to properly catch the missing label condition in the release notes generation.

**Comment by jesusvasquez333** [comment]
I ended up writing a new GitHub action to work around this issue: https://github.com/marketplace/actions/verify-pull-request-labels. 

Instead of relying on the Check status, it uses pull request reviews request instead. I will open a PR now.

This will make sure a PR has at least one of the mandatory labels. But the fact that labels can be added/deleted on a PR after it has been merged remains. But at least that is less likely to happen.
